src/main/java/ch/njol/skript/expressions/ExprFurnaceSlot.java

```diff
@@ -156,104 +139,20 @@ public Class<? extends Slot> getReturnType() {
+		}
 		if (isEvent) {
-			return "the " + time + slotName + (isResultSlot ? " slot" : "");
+			result += " of " + event.getEventName();
 		} else {
-			assert blocks != null;
-			return "the " + time + slotName + " slot of " + blocks.toString(event, debug);
+			result += " of " + blocks.toString(event, debug);
 		}
+		return result;
 	}
 
-	@Override
-	public boolean setTime(int time) {
-		if (isEvent) { // getExpr will be null
-			if (slot == RESULT && !isResultSlot) { // 'the past/future result' - doesn't make sense, don't allow it
-				return false;
-			} else if (slot == FUEL) {
-				return setTime(time, FurnaceBurnEvent.class);
-			} else {
-				return setTime(time, FurnaceSmeltEvent.class);
-			}
-		}
-		return false;
-	}
-	
-	private final class FurnaceEventSlot extends InventorySlot {
-		
-		private final Event event;
-		
-		public FurnaceEventSlot(Event event, FurnaceInventory furnaceInventory) {
-			super(furnaceInventory, slot);
-			this.event = event;
-		}
-		
-		@Override
-		@Nullable
-		public ItemStack getItem() {
-			switch (slot) {
-				case ORE:
-					if (event instanceof FurnaceSmeltEvent) {
-						ItemStack source = ((FurnaceSmeltEvent) event).getSource().clone();
-						if (getTime() != EventValues.TIME_FUTURE)
-							return source;
-						source.setAmount(source.getAmount() - 1);
-						return source;
-					}
-					return super.getItem();
-				case FUEL:
-					if (event instanceof FurnaceBurnEvent) {
-						ItemStack fuel = ((FurnaceBurnEvent) event).getFuel().clone();
-						if (getTime() != EventValues.TIME_FUTURE)
-							return fuel;
-						// a single lava bucket becomes an empty bucket
-						// see https://minecraft.wiki/w/Smelting#Fuel
-						// this is declared here because setting the amount to 0 may cause the ItemStack to become AIR
-						Material newMaterial = fuel.getType() == Material.LAVA_BUCKET ? Material.BUCKET : Material.AIR;
-						fuel.setAmount(fuel.getAmount() - 1);
-						if (fuel.getAmount() == 0)
-							fuel = new ItemStack(newMaterial);
-						return fuel;
-					}
-					return super.getItem();
-				case RESULT:
-					if (event instanceof FurnaceSmeltEvent) {
-						ItemStack result = ((FurnaceSmeltEvent) event).getResult().clone();
-						if (isResultSlot) { // Special handling for getting the result slot
-							ItemStack currentResult = ((FurnaceInventory) getInventory()).getResult();
-							if (currentResult != null)
-								currentResult = currentResult.clone();
-							if (getTime() != EventValues.TIME_FUTURE) { // 'past result slot' and 'result slot'
-								return currentResult;
-							} else if (currentResult != null && currentResult.isSimilar(result)) { // 'future result slot'
-								currentResult.setAmount(currentResult.getAmount() + result.getAmount());
-								return currentResult;
-							} else {
-								return result;
-							}
-						}
-						// 'the result'
-						return result;
-					}
-					return super.getItem();
-			}
-			return null;
-		}
```

Comment:
can you give me a quick explanation on why you removed this?

Reply:
From my understanding of it, it was primarily used to work inside of events to get and/or change their respective items. However, this has been moved into mine,

Reply:
Have you tested all the permutations of past/future to ensure it still works as it did before?